Foxy is now EOL

moriarty opened the pull request:
It is June 1 2023 and Foxy has reached EOL

moriarty committed:
Foxy is now EOL

It is June 1 2023 and Foxy has reached EOL

Signed-off-by: Alex Moriarty <alex.moriarty@picknik.ai>

fujitatomoya reviewed (changes requested):
@moriarty thanks for the contribution.

can you add the following changes?

- `.mergify.yml` foxy backport rule can be removed. (refer to https://github.com/ros2/ros2_documentation/pull/3243)
- https://github.com/ros2/ros2_documentation/blob/6de7fb9b69cad003f1301dfdc2bd6f3df158d898/source/Releases.rst?plain=1#L52 should be removed to turn the background into gray.
- https://github.com/ros2/ros2_documentation/blob/6de7fb9b69cad003f1301dfdc2bd6f3df158d898/source/Releases.rst?plain=1#L109 should be updated to exact E.O.L date.

fujitatomoya commented:
@cottsay @audrow @Yadunund can you have another review?
CC: @clalancette

clalancette commented:
We have not officially EOLed Foxy yet, so we should hold off on this.  It is coming soon.

moriarty commented:
> We have not officially EOLed Foxy yet, so we should hold off on this. It is coming soon.

Why is it not official yet?

The original EOL date has passed.

clalancette commented:
> Why is it not official yet?

We haven't done the final sync.  The work is in progress.  In general, we don't guarantee an EOL date, just have an approximate end.

moriarty committed:
[EOL] Foxy

- review feedback:
  - mergify config
  - background color
  - add date to date

Signed-off-by: Alex Moriarty <alex.moriarty@picknik.ai>

moriarty commented:
> > Why is it not official yet?
> 
> We haven't done the final sync. The work is in progress. In general, we don't guarantee an EOL date, just have an approximate end.

I've set the date in that line to the last day of May 2023.
It seems odd to extend the EOL into June after it's been documented for a long time as May 2023.

https://www.ros.org/reps/rep-2000.html#foxy-fitzroy-may-2020-may-2023

For any projects which want to support current EOL versions this is effects things like resource planning and scheduling, and continuous integration budgets. I understand that the sync sometimes happens when the sync happens and that the last sync can easily get delayed due to Iron, but other non-core ROS 2 projects have already started assuming Foxy is EOL and dropping CI etc

clalancette commented:
> I understand that the sync sometimes happens when the sync happens and that the last sync can easily get delayed due to Iron

Yes, that's generally the reason.  It's not impossible to do it at the same time as releasing a new release, but it can be hard at times.  That was the case this year.

moriarty commented:
used GitHub UI to fixup conflicts and remove .rst extension from #3723

moriarty commented on source/Releases.rst:
```diff
@@ -106,7 +105,7 @@ Rows in the table marked in green are the currently supported distributions.
      - June 5th, 2020
      - |foxy|
-     - May 2023
+     - May 31st, 2023
```
```suggestion
     - June 20th, 2023
```

moriarty committed:
Update source/Releases.rst

Official ROS 2 Foxy EOL Date June 20th, 2023

https://twitter.com/OpenRoboticsOrg/status/1666484720585277443

moriarty commented:
@clalancette I've updated the date to match https://twitter.com/OpenRoboticsOrg/status/1666484720585277443

moriarty commented:
> @moriarty thanks for the contribution.
> 
> can you add the following changes?
> 
> * `.mergify.yml` foxy backport rule can be removed. (refer to [Mark ROS 2 Galactic as end-of-life #3243](https://github.com/ros2/ros2_documentation/pull/3243))
> * https://github.com/ros2/ros2_documentation/blob/6de7fb9b69cad003f1301dfdc2bd6f3df158d898/source/Releases.rst?plain=1#L52
>    should be removed to turn the background into gray.
> * https://github.com/ros2/ros2_documentation/blob/6de7fb9b69cad003f1301dfdc2bd6f3df158d898/source/Releases.rst?plain=1#L109
>    should be updated to exact E.O.L date.

I Applied your feedback, could you re-review?

fujitatomoya commented on source/Releases.rst:
```diff
@@ -106,7 +105,7 @@ Rows in the table marked in green are the currently supported distributions.
      - June 5th, 2020
      - |foxy|
-     - May 2023
+     - June 20th, 2023
```
This needs to be fixed yet? https://discourse.ros.org/t/new-packages-for-foxy-fitzroy-2023-06-20/32039 still under process for final sync. everything else looks good to me. we need a final E.O.L date.

moriarty replied:
Both the link you provide and the official open robotics Twitter account point to June 20th, 2023…

what date are you suggesting is the final EOL date?

https://twitter.com/OpenRoboticsOrg/status/1666484720585277443

clalancette replied:
Yes, we are officially at end-of-life now, so we can put this in.

I've gone ahead and rebased it and made one more fix.  Assuming CI comes back clean, I'll go ahead and merge this.  Thanks for the PR.

clalancette committed:
Remove Foxy from the top-level Releases page.

Signed-off-by: Chris Lalancette <clalancette@gmail.com>

clalancette commented:
@Mergifyio backport foxy
